fairing/kubernetes/manager.py

```diff
@@ -56,6 +56,40 @@ def delete_deployment(self, name, namespace):
             namespace,
             client.V1DeleteOptions())
 
+    def secret_exists(self, name, namespace):
+        secrets = client.CoreV1Api().list_namespaced_secret(namespace)
+        secret_names = [secret.metadata.name for secret in secrets.items]
+        return name in secret_names
+
+    def add_credentials_to_pod_spec(self, pod_spec):
```

Comment:
This method doesn't require the kubernetes API endpoint so it doesn't have to be on the Manager class and can work by itself

Reply:
Moved to `fairing/cloud/gcp.py`

Also updated the function to take as input the kubernetes manager and namespace, so we can check if the secret exists in the function itself.